### PR TITLE
rework generic modal styling to not clash with other style rules

### PR DESCRIPTION
**Pull request by madsWesth**
Changed the styling to not have a separate stylesheet with custom class names. The styles in the stylesheet had nothing wrong with them and were done well. I changed it because:
- Class names conflicted with current setup and changed styles outside of the component.
- To rely more on daisyUI and tailwind for code consistency(no other place had a stylesheet) and to have changes reflected in the component if we make changes to tailwind/daisyUI theme.

I ran into issues with login at the end while trying to verify that everything looked okay, so this still needs to be checked

Cc: @MartinJenzen 
Nice to see people actually using and improving generic components instead of making new components that arent reusable 👍

### CODE REVIEW

**Comment by MartinJenzen** [comment]
@madsWesth It is wonderful with such a nice reaction and quick response, and thanks for the compliment by the way👍

But a small request, can you align your redesign to match more with the original Figma-aligned design, please? 😁
- Close button (X) is not on the same height as modal title, and not right aligned with confirm button.
- Distance between body text and action buttons are less than Figma design.
- Cancel button should be pushed a bit left to match with the left edge of the body text.

Your redesign:
image
With hover effect:
image

Original design (Figma-aligned):
image
With hover effect (a bit difficult to see side-by-side, but hover effects is: bg-primary -> bg-cyan-900
image

I will review PR when all this is corrected 😉 Maybe you can review my PR (#360 - Mobile) in return? 😁

File: app/src/components/GenericModalComponent.tsx [hunk 1]
```diff
@@ -44,46 +44,59 @@ const GenericModalComponent: React.FC<GenericModalProps> = ({
...
-            {cancelBtnText}
+          <button
+            type="button"
+            className="flex flex-col w-40 btn bg-transparent border-none hover:bg-transparent px-0"
+            onClick={onClose}
+          >
+            <span className="text-[#166276] normal-case text-lg">
+              {cancelBtnText}
+            </span>
+            <hr className="w-3/5 border-[#166276]" />
```
Comment (MartinJenzen) [inline app/src/components/GenericModalComponent.tsx:85]: Remember to use the custom Tailwind CSS class 'bg-primary' here instead of the hex-color code ;)

File: app/src/components/styles/GenericModalComponent.css
Comment (MartinJenzen) [inline app/src/components/styles/GenericModalComponent.css:1]: Auch! Dangerous to delete this styling component, as it may break the modal everywhere else it is being used 😬

**Comment by MartinJenzen** [comment]
@madsWesth 

Possible solution: 
- Revert back to original version of GenericModalComponent.tsx and .css
- Adapt .tsx to accept an optional argument that will trigger conditionals that will then provide and/or suppress the behavior you need and/or don't need in your particular modal.
- Add new classes to the .css that support your particular needs.

File: app/src/components/styles/GenericModalComponent.css
Comment (madsWesth) [reply app/src/components/styles/GenericModalComponent.css:1]: Not used anywhere other than the modal component as far as i can tell

File: app/src/components/GenericModalComponent.tsx (hunk 1 above)
Comment (madsWesth) [reply app/src/components/GenericModalComponent.tsx:85]: True! I didnt even think of that, I might just do a find and replace all on the entire repo tbh 😆

**Comment by MartinJenzen** [comment]
@madsWesth Be advised that @Todua992 has fixed this issue, among other things, on PBI #194. So I guess we can cancel this PR 🤷🏻‍♂️

**Comment by madsWesth** [comment]
> @madsWesth Be advised that @Todua992 has fixed this issue, among other things, on PBI #194. So I guess we can cancel this PR 🤷🏻‍♂️

I heavily disagree with his approach, so i will keep this open.

**Review by Todua992 — APPROVED** [review]
All looks good for me, you are welcome to put this changes to the Branch.

**Comment by madsWesth** [comment]
@MartinJenzen if you want to re review that would be nice. I think i addressed what you brought up, let me know if there are places you think need additional styling based on some conditions or if anything is missing.

In regards to using CSS files instead of purely using tailwind classes, i personally prefer that approach. On the other hand i would rather have an inferior approach all throughout the project than a mix of a good and a bad one. I think that is better for the health of the project. 
I would be fully down for a change in approach in how we do styling, but i believe that it should be done properly so everybody follows that standard and it is enforced so there isnt a mix of both where both ways are used.
The current issue with way too many class names on everything should hopefully be fixed soon with a theme made for daisyUI that makes most classes redundant. I created an issue for this a while back but havent had the time to actually do it https://github.com/Educado-App/educado-frontend/issues/168

And thanks for all the good feedback 😄

**Review by MartinJenzen — CHANGES_REQUESTED** [review]
This approach totally massacres an otherwise elegant and clean approach. No bueno!

**Comment by madsWesth** [comment]
> @madsWesth I **absolutely** wholeheartedly disagree with your reasoning and approach! A _bad_ practice having been followed throughout the project does _not_ excuse continuing with that bad practice! The change has to start somewhere, and _we_ are responsible that change!
> 
> My approach with `Tailwind CSS` properties in a separate styling file `GenericModalComponent.css` is much more readable and clean than having a million of these properties defined throughout the HTML-code.
> 
> You really didn't have to massacre `GenericModalComponent.tsx` let alone delete `GenericModalComponent.css` to make it fit with your needs. There was a much more simple and elegant way to achieve this than to _butcher_ the code, resulting in me having my time and effort needlessly wasted. An alternative and better option would have been to not touch the generic modal at all and made your own specialized modal instead.
> 
> I expect you to **not** follow through with these changes!

TLDR; Not my intention to massacre, you're welcome to change it to use a css file.
There are loads of fixes and changes there wasn't in the original styling so that would have to be re-adapted.

Completely fair we are both entitled to our opinions and its okay to disagree. I thought that it was old/legacy code from a time before a standard had been developed instead of something you recently made and therefore was more "fair game" to modify, that's my fault. My intention was not overrule any decision you had made, it was also pretty clear that you had spend some time understanding the tech and crafting that solution (unlike most people that just copilot it) so i can understand that it probably feels pretty bad to see it removed that way.

I do still disagree with you and i don't think it's a good approach currently, and that isn't because I'm a fan of tailwind like you i find it kinda hard to read, the main reason being that it deviates from the standard of the codebase.
Deviating isn't inherently good or bad, but standards do have a lot of value! So when deviations from a standard are made they should be in a pursuit of setting a new standard.

A good rule of thumb is that "the nail that sticks out gets hammered down", so if you haven't made an extra effort to make sure the deviation sticks and becomes the new "approach"/standard, it will most likely get the hammer, and usually pretty quickly, which is what happened here.

If you really want to try and set a new standard you're more than welcome, I'm not and expert on tailwind so i don't know what is considered good practice so wont debate with you on that. Trying to make change is hard though so urge you to think about how you're gonna get other people adopt this and maybe some type of enforcement so people don't ignore it anyway. When you make a change like this you also kinda assume the responsibility for doing this.

You're more than welcome to keep the styling in a separate css file, I'm not gonna stop you. Beware if that other people might also do the same thing I did, if there is nothing in place to deter them 😅 There are some bug fixes and other changes that have been made in this PR that would need to be adopted. Also there would need to be some scheme for naming the classes so they don't conflict with the base styles of tailwind or daisyUI which was the reason this even became an issue in the first place.

I'm not emotionally attached to a specific standard so i don't care what we do at the end of the day, I just want well defined standards and guidelines for the cohesion of the project, which was my main motivation for making this change.

I don't like that you only bring this up now. You didnt object initially, so i didn't think there was an issue and kept spending time on this which could have been used better. This wasn't done to insult you or the work you had done, because that was all high quality and well thought out, so no offense meant!
